src/services/textChanges.ts

```diff
@@ -483,6 +483,7 @@ export class ChangeTracker {
     private readonly newFiles: { readonly oldFile: SourceFile | undefined, readonly fileName: string, readonly statements: readonly (Statement | SyntaxKind.NewLineTrivia)[] }[] = [];
     private readonly classesWithNodesInsertedAtStart = new Map<number, { readonly node: ClassLikeDeclaration | InterfaceDeclaration | ObjectLiteralExpression, readonly sourceFile: SourceFile }>(); // Set<ClassDeclaration> implemented as Map<node id, ClassDeclaration>
     private readonly deletedNodes: { readonly sourceFile: SourceFile, readonly node: Node | NodeArray<TypeParameterDeclaration> }[] = [];
+    private readonly changeExistingFile: { readonly oldFile: SourceFile, readonly newFile: SourceFile | undefined, readonly statements: readonly (Statement | SyntaxKind.NewLineTrivia)[] }[] = [];

```

Comment:
You’re still planning to delete the `textChanges` changes, right?

Reply:
I've deleted `changeExistingFile` but I needed to differentiate between 'Move to file' and 'Move to a new file' so added an optional parameter in `getTextChangesFromChanges` so that it uses the correct sourceFile to generate the `newText`.

Reply:
I’m not following why `textChanges` needs any changes since this refactor is just editing multiple files, which `textChanges` already supports in many other refactors. Maybe you can walk me through it on a call?